### PR TITLE
Associate room with creator

### PR DESCRIPTION
**Pull request by isisblack**
- Add newuser.html and newuser.js
     - newuser.html: has create room button & join room form
     - newuser.js: now holds create room function
- RoomServlet.java adds user that created the room to list of users in that room
- Print current room url on lobby page => https://screenshot.googleplex.com/LRfeYhGuhpQ.png

"Create list of users, Java formatting" commit made changes to many files after running java formatter

### CODE REVIEW

File: src/main/java/com/google/songgame/servlets/RoomServlet.java [hunk 1]
```diff
@@ -41,17 +42,58 @@ public void init() {
...
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Room");
+    PreparedQuery results = datastore.prepare(query);
+    
+    Map<String, String[]> roomIdParamMap = request.getParameterMap();
+
+    for (String roomId : roomIdParamMap.keySet()) {
+    String[] idArray = (String[]) roomIdParamMap.get(roomId);
```
Comment (salnad) [inline src/main/java/com/google/songgame/servlets/RoomServlet.java]: this line should be indented (try running the Java formatter, might catch other formatting things I missed(

File: src/main/java/com/google/songgame/servlets/RoomServlet.java [hunk 2]
```diff
@@ -41,17 +42,58 @@ public void init() {
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Map<String, String> roomProperties = readJSONFromRequest(request);
 
-    // Save player roomId to datastore.
+    // Save roomId to datastore.
     Entity roomEntity = new Entity("Room");
     roomEntity.setProperty("roomId", roomProperties.get("roomId"));
+    roomEntity.setProperty("userId", roomProperties.get("userId"));
```
Comment (salnad) [inline src/main/java/com/google/songgame/servlets/RoomServlet.java:48]: Is there any way to store multiple userId's? You might want to look into storing a list of userId's (you can store ArrayList in Datastore)

File: src/main/webapp/index.html [hunk 3]
```diff
@@ -2,7 +2,7 @@
 <html>
   <head>
     <meta charset="UTF-8">
-    <title>SOng Guessing Game</title>
+    <title>Song Guessing Game</title>
     <script src="room.js"></script>
     <script src="game.js"></script>
```
Comment (salnad) [inline src/main/webapp/index.html]: Unnecessary javascript file (you can remove this)

File: src/main/webapp/room.js [hunk 4]
```diff
@@ -18,29 +18,10 @@ const createUser = async (ev) => {
       });
 
     // Sends user to lobby.html on button click.
```
Comment (salnad) [inline src/main/webapp/room.js:20]: Change comment (currently says lobby.html)

File: src/main/java/com/google/songgame/servlets/RoomServlet.java [hunk 5]
```diff
@@ -36,22 +37,44 @@ public void init() {
...
 
-    // Save player roomId to datastore.
+    List<String> userIdList = new ArrayList<String>();
+    userIdList.add(roomProperties.get("userId"));
+
+    // Save roomId to datastore.
     Entity roomEntity = new Entity("Room");
     roomEntity.setProperty("roomId", roomProperties.get("roomId"));
-    
+    roomEntity.setProperty("userId", userIdList);
```
Comment (salnad) [inline src/main/java/com/google/songgame/servlets/RoomServlet.java:51]: Might want to change the name of this property to userIdList instead of keeping it as userId

File: src/main/java/com/google/songgame/data/YoutubeParser.java [hunk 6]
```diff
@@ -116,9 +111,9 @@ private static PlaylistItemListResponse getPlaylistInfo(String playlistId)
   }
   /**
    * Return a Video object chosen at random from a given list of videoIds
-   * 
-   * <p>Choose a videoId at random from a given ArrayList of videoIds and use the YouTube Data API 
-   * to return the Video Object with that particular videoId 
+   *
+   * <p>Choose a videoId at random from a given ArrayList of videoIds and use the YouTube Data API
```
Comment (pruess-google) [inline src/main/java/com/google/songgame/data/YoutubeParser.java:115]: Looks like this line is over the limit. Could you fix this?

File: src/main/webapp/lobby.js [hunk 7]
```diff
@@ -38,3 +38,5 @@ function createUsernameElement(username) {
     node.appendChild(textnode);
     return node;
   }
+
+document.getElementById("url").innerHTML = "Share the link!:<br>" + (window.location.href);
```
Comment (pruess-google) [inline src/main/webapp/lobby.js:43]: Oh interesting, does this display the link in the lobby for other users to share? Could you include a screenshot of this in your PR description?

File: src/main/webapp/newuser.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (pruess-google) [inline src/main/webapp/newuser.html:1]: I was wondering why you're choosing to call this "newuser.html", since it looks like this page allows a user to either join an existing room or create a new room, so I don't think it has to do with creating a new user.

I think a better name might be "room.html" or "join-room.html". What do you think?

The same applies to the JS file as well

File: src/main/webapp/room.js [hunk 9]
```diff
@@ -17,30 +17,11 @@ const createUser = async (ev) => {
...
-  await fetch("/room", {
-    method: "POST",
-    body: JSON.stringify(room),
-    headers: {
-      "Content-Type": "application/json",
-    },
-  });
-  
+    // Sends user to newuser.html on button click.
+    window.location.href = 'newuser.html';
```
Comment (pruess-google) [inline src/main/webapp/room.js:21]: Similar to what I said about "newuser.html", this file doesn't seem to be related to rooms anymore. Instead, it seems like it allows you to create a new user. So maybe this could be called "newuser.js" or just "user.js".

What do you think?

File: src/main/webapp/lobby.js (hunk 7 above)
Comment (alekseychaikovsky) [reply src/main/webapp/lobby.js:43]: +1 would love to see this.

File: src/main/webapp/room.js (hunk 9 above)
Comment (isisblack) [reply src/main/webapp/room.js:21]: @pruess-google sounds good, I changed it to user.js!

File: src/main/webapp/newuser.html (hunk 8 above)
Comment (isisblack) [reply src/main/webapp/newuser.html]: @pruess-google yes I thinking of it as the user either joining or creating a room which makes them a new user, but that doesn't really make any sense. But join-room makes a lot of sense, I changed it!
